fractal_server/zip_tools.py

```diff
@@ -0,0 +1,86 @@
+import os
+import shutil
+from io import BytesIO
+from pathlib import Path
+from zipfile import ZIP_DEFLATED
+from zipfile import ZipFile
+
+
+def _zip_folder_to_byte_stream(*, folder: str):
+    """
+    Get byte stream with the zipped log folder of a job.
+
+    Args:
+        folder: the folder to zip
+    """
+    zip_file = Path(f"{folder}.zip")
+
+    if os.path.exists(zip_file):
+
+        def iterfile():
+            """
+            https://fastapi.tiangolo.com/advanced/
+            custom-response/#using-streamingresponse-with-file-like-objects
+            """
+            with open(zip_file, mode="rb") as file_like:
+                yield from file_like
+
+        return iterfile()
+
+    else:
+
+        byte_stream = BytesIO()
+        with ZipFile(
+            byte_stream, mode="w", compression=ZIP_DEFLATED
+        ) as zipfile:
+            for root, dirs, files in os.walk(folder):
+                for file in files:
+                    file_path = os.path.join(root, file)
+                    archive_path = os.path.relpath(file_path, folder)
+                    zipfile.write(file_path, archive_path)
+        return iter([byte_stream.getvalue()])
+
+
+def _zipping_worked(*, folder: str) -> bool:
+
+    zip_file = f"{folder}_tmp.zip"
+
+    # CHECK 1: zip file exists
+    if not os.path.exists(zip_file):
+        return False
+
+    folder_size = 0
+    folder_files = set()
+    for dirpath, dirnames, filenames in os.walk(folder):
+        for f in filenames:
+            fp = os.path.join(dirpath, f)
+            folder_files.add(os.path.relpath(fp, folder))
+            folder_size += os.path.getsize(fp)
+
+    with ZipFile(zip_file, "r") as zip_ref:
+        zip_files = set(
+            name for name in zip_ref.namelist() if not name.endswith("/")
+        )
+    zip_size = os.path.getsize(zip_file)
+
+    # CHECK 2: all files are there
+    if folder_files != zip_files:
+        return False
+
+    # CHECK 3: zip file is at least 10% of the original folder
+    if zip_size * 10 < folder_size:
```

Comment:
Make absolute rather than relative. The idea is that when the zip is tiny either it's not relevant to remove the folder or something may have gone wrong.

Threshold: e.g. 1M?